src/core/lib/resource_quota/memory_quota.h

```diff
@@ -299,7 +299,7 @@ class GrpcMemoryAllocatorImpl final : public EventEngineMemoryAllocatorImpl {
     // from  0 to non-zero, then we have more to do, otherwise, we're actually
     // done.
     size_t prev_free = free_bytes_.fetch_add(n, std::memory_order_release);
-    if (prev_free + n > kMaxQuotaBufferSize) {
+    if (donate_back_.Tick()) {
```

Comment:
Should we have both conditions here: i.e, if (donate_back_.Tick() || prev_free + n > kMaxQuotaBufferSize) ?

Reply:
done